Fixes #30348 - trends cleanup task

ezr-ondrej opened the pull request:
Cleans up trends data for users who wish to stop using trends.

ezr-ondrej committed:
Fixes #30348 - trends cleanup task

tbrisker commented on lib/tasks/trends.rake:
```diff
@@ -1,4 +1,16 @@
+    end
+    FakeTrendCounter.delete_all
+    FakeTrend.delete_all
+  end
```
We need to also cleanup the db tables, permissions and settings when running this task. Should we put it in a different namespace where we can put additional cleanup tasks in the future?

ezr-ondrej replied:
My :heart: hurts handling schema anywhere else then migrations. Do we really want to drop the tables from the task? Given there should be a migration that will drop all the data for everyone.
Setting and perms I agree :100:  :+1:

tbrisker replied:
I'm not sure about dropping it for everyone in migration, what if i upgraded and didn't mean to remove the data but only realized that i need to install the plugin too late?

ezr-ondrej replied:
well that's a bummer and tough life but that is what it is... You've had two versions to realize that.
I mean assuming we are going to add the migration with droping the deprecated endpoints.

tbrisker replied:
I think it makes sense to drop here as well - when someone runs this task, they are saying "we don't want this stuff, get rid of it", which to me means we should also drop the tables. We can still add a migration as well to make sure we remove them when we drop the api endpoints for those who were lazy and didn't care about running the task.

ekohl commented on lib/tasks/trends.rake:
```diff
@@ -1,4 +1,16 @@
 namespace :trends do
+  desc 'Clean up Trends data'
+  task :clobber => :environment do
```
`clobber` doesn't tell me exactly what it'd do. How about `purge`?

ekohl commented:
There is still a cronjob:

https://github.com/theforeman/foreman-packaging/blob/56cebd1c4617d67b0e7a9f33fb5541f0b4b1e50e/packages/foreman/foreman/foreman.cron.d#L18-L19
https://github.com/theforeman/foreman-packaging/blob/caa5b6b984d60da8f9eee2b77a4e589376b3e2e4/debian/bionic/foreman/foreman.cron.d#L18-L19
https://github.com/theforeman/foreman-packaging/blob/caa5b6b984d60da8f9eee2b77a4e589376b3e2e4/debian/buster/foreman/foreman.cron.d#L18-L19

With that present, it'll quickly be populated again. Don't forget to clean that up as well.

tbrisker commented:
@ezr-ondrej those cron jobs probably need to move in packaging to the plugin's package

ezr-ondrej commented:
I've moved the cron jobs to a Dynflow action: https://github.com/theforeman/foreman_statistics/pull/19
I've moved the task to the namespace `purge` and introduced new rake `purge_data`, that can be used for future purges and that should cleanup all unnecessary data.

ezr-ondrej commented:
I've deleted the tables. Just a note to consider, the plugin is not installable after run of this script in current setting.
Should we squash migrations in plugin then?

ezr-ondrej commented:
@tbrisker do you have more comments?

tbrisker commented:
@ezr-ondrej could we somehow also delete the migrations from the db? that way if the plugin is installed after this task is run it will think those migrations never executed.
Also, still need to clean up permissions and settings.

ezr-ondrej commented:
Rake is dropping Permission, Setting and removes records in the SchemaMigration now.

tbrisker reviewed (approved):
Thanks @ezr-ondrej !

tbrisker reviewed (approved):
tested, dropping works and so does re-migrating afterwards with the plugin installed.

tbrisker commented:
2.2 - b01ab75b2

tbrisker commented:
I just realized there's also permissions for statistics that we should clean up - could you please add a follow up pr @ezr-ondrej ?

ezr-ondrej commented:
Sure: #7932
